engine-nextgen/src/main/kotlin/graphql/nadel/enginekt/transform/skipInclude/SkipIncludeTransform.kt

```diff
@@ -0,0 +1,125 @@
+package graphql.nadel.enginekt.transform.skipInclude
+
+import graphql.execution.MergedField
+import graphql.introspection.Introspection
+import graphql.language.Field
+import graphql.nadel.Service
+import graphql.nadel.ServiceExecutionResult
+import graphql.nadel.enginekt.NadelExecutionContext
+import graphql.nadel.enginekt.blueprint.NadelOverallExecutionBlueprint
+import graphql.nadel.enginekt.transform.NadelTransform
+import graphql.nadel.enginekt.transform.NadelTransformFieldResult
+import graphql.nadel.enginekt.transform.artificial.NadelAliasHelper
+import graphql.nadel.enginekt.transform.query.NadelQueryTransformer
+import graphql.nadel.enginekt.transform.result.NadelResultInstruction
+import graphql.nadel.enginekt.transform.skipInclude.SkipIncludeTransform.State
+import graphql.nadel.enginekt.util.resolveObjectTypes
+import graphql.nadel.enginekt.util.toBuilder
+import graphql.normalized.ExecutableNormalizedField
+import graphql.normalized.ExecutableNormalizedField.newNormalizedField
+import graphql.schema.GraphQLSchema
+
+internal class SkipIncludeTransform : NadelTransform<State> {
+    companion object {
+        private const val skipFieldName = "__skip"
+    }
+
+    class State(
+        val aliasHelper: NadelAliasHelper,
+    )
+
+    override suspend fun isApplicable(
+        executionContext: NadelExecutionContext,
+        executionBlueprint: NadelOverallExecutionBlueprint,
+        services: Map<String, Service>,
+        service: Service,
+        overallField: ExecutableNormalizedField,
+    ): State? {
+        // This hacks together a child that will pass through here
+        if (overallField.children.isEmpty()) {
+            val mergedField = executionContext.query.getMergedField(overallField)
+            if (hasAnyChildren(mergedField)) {
```

Comment:
So the normalised field builder automatically removes fields that resolve to `@skip(if: true)`. So we check if the field is currently empty, then check if the original merged field _had_ children